<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsUI</class>
 <widget class="QWidget" name="settingsWidget">
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="generalTab">
       <attribute name="title">
        <string>General</string>
       </attribute>
       <!-- You can add layouts and widgets to this tab here -->
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="outputMonitorGroupBox">
          <property name="title">
           <string>Output Monitor</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label_output_monitor">
             <property name="text">
              <string>Select Monitor:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="monitorSelectionComboBox"/>
           </item>
           <item>
            <widget class="QPushButton" name="refreshMonitorsButton">
             <property name="text">
              <string>Refresh</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item> <!-- DeckLink Output GroupBox -->
         <widget class="QGroupBox" name="decklinkOutputGroupBox">
          <property name="title">
           <string>DeckLink Output</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_decklink">
            <!-- Row 0: Fill Device -->
           <item row="0" column="0"> <!-- Label for Fill Device -->
            <widget class="QLabel" name="label_decklink_fill_device">
             <property name="text">
              <string>Fill Device:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1"> <!-- ComboBox for Fill Device -->
            <widget class="QComboBox" name="decklinkFillDeviceComboBox"/>
           </item>
           <item row="0" column="2"> <!-- Refresh Button (spans 2 rows for alignment or keep separate) -->
            <widget class="QPushButton" name="refreshDecklinkDevicesButton">
             <property name="text">
              <string>Refresh</string>
             </property>
             <property name="toolTip">
              <string>Refresh the list of available DeckLink devices.</string>
             </property>
            </widget>
           </item>
            <!-- Row 1: Key Device -->
           <item row="1" column="0"> <!-- Label for Key Device -->
            <widget class="QLabel" name="label_decklink_key_device">
             <property name="text">
              <string>Key Device:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1"> <!-- ComboBox for Key Device -->
            <widget class="QComboBox" name="decklinkKeyDeviceComboBox"/>
           </item>
            <!-- Row 2: Video Mode (was row 1) -->
           <item row="2" column="0">
            <widget class="QLabel" name="label_decklink_video_mode">
             <property name="text">
              <string>Video Mode:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="decklinkVideoModeComboBox">
             <property name="enabled">
              <bool>false</bool> <!-- Initially disabled until a device is selected and modes are loaded -->
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>        <item>
         <spacer name="verticalSpacer_general">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="slideDefaultsTab">
       <attribute name="title">
        <string>Slide Defaults</string>
       </attribute>
       <layout class="QVBoxLayout" name="slideDefaultsLayout">
        <item>
         <widget class="QGroupBox" name="newSlideDefaultGroupBox">
          <property name="title">
           <string>New Slide Default Template</string>
          </property>
          <layout class="QHBoxLayout" name="newSlideDefaultHLayout">
           <item>
            <widget class="QLabel" name="label_defaultTemplate">
             <property name="text">
              <string>Default Template:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="defaultTemplateComboBox"/>
           </item>
           <item>
            <spacer name="horizontalSpacer_slideDefaults">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_slideDefaults">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="developerTab">
       <attribute name="title">
        <string>Developer</string>
       </attribute>
       <!-- You can add layouts and widgets to this tab here -->
       <layout class="QVBoxLayout" name="developerTabLayout">
       <item>
         <widget class="QGroupBox" name="ProdToggleGroupBox">
          <property name="title">
           <string>Prod Toggle</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label_prod_toggle">
             <property name="text">
              <string>Version:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="ProdToggleComboBox"/>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_general">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum> <!-- Make this spacer explicitly Fixed -->
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QGroupBox" name="benchmarkingGroupBox">
          <property name="title">
           <string>Benchmarking</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>false</bool> <!-- Collapsed by default -->
          </property>
          <layout class="QVBoxLayout" name="benchmarkingLayout">
           <item>
            <widget class="QLabel" name="label_app_init_time">
             <property name="text">
              <string>App Init Time: N/A</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_mw_init_time">
             <property name="text">
              <string>MainWindow Init Time: N/A</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_mw_show_time">
             <property name="text">
              <string>MainWindow Show Time: N/A</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label_last_presentation_path">
             <property name="text">
              <string>Last Presentation: None</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_pm_load_time">
             <property name="text">
              <string>PM Load Time: N/A</string>
             </property>
             <property name="toolTip">
              <string>Time taken by the Presentation Manager to load the presentation file from disk and parse its content.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_ui_update_time">
             <property name="text">
              <string>UI Update Time: N/A</string>
              </property>
             <property name="toolTip">
              <string>Time taken to update the main window's UI after a presentation is loaded, including rendering all slide previews.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_render_total">
             <property name="text">
              <string>Render (Total): N/A</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_render_images">
             <property name="text">
              <string>Render (Images): N/A</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_render_fonts">
             <property name="text">
              <string>Render (Fonts): N/A</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_render_layout">
             <property name="text">
              <string>Render (Layout): N/A</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_render_draw">
             <property name="text">
              <string>Render (Draw): N/A</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>0</height> <!-- Ensure this expanding spacer desires 0 initial height -->
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="backupSharingTab">
       <attribute name="title">
        <string>Backup &amp; Sharing</string>
       </attribute>
       <layout class="QVBoxLayout" name="backupSharingLayout">
        <item>
         <widget class="QLabel" name="backupStatusLabel">
          <property name="text">
           <string>Status: Not Configured</string>
          </property>
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="unconfiguredRepoWidget" native="true">
          <layout class="QVBoxLayout" name="unconfiguredRepoLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="backupIntroLabel">
             <property name="text">
              <string>Configure a Git repository to enable backup and sharing of your data. This allows you to keep your data synchronized across multiple devices or restore it easily.</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
             <property name="margin">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="configureButtonLayout">
             <item>
              <spacer name="horizontalSpacer_configure_left">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="setupNewRepoButton"> <!-- Renamed from configureRepoButton -->
               <property name="text">
                <string>Setup New Backup Location...</string> <!-- Changed text -->
               </property>
               <property name="minimumSize">
                <size>
                 <width>200</width>
                 <height>0</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Set up a new backup location (creates a new bare Git repository).</string> <!-- Updated tooltip -->
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_configure_right">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QGroupBox" name="existingRepoGroupBox">
             <property name="title">
              <string>Or Connect to Existing Remote Repository</string>
             </property>
             <layout class="QFormLayout" name="existingRepoFormLayout">
              <item row="0" column="0">
               <widget class="QLabel" name="label_existingRepoUrl">
                <property name="text">
                 <string>Remote URL/Path:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="existingRepoUrlLineEdit">
                <property name="placeholderText">
                 <string>e.g., https://github.com/user/repo.git or /path/to/remote.git</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <layout class="QHBoxLayout" name="connectExistingButtonLayout">
                <item>
                 <spacer name="horizontalSpacer_connect_existing_left">
                  <property name="orientation"><enum>Qt::Horizontal</enum></property>
                  <property name="sizeType"><enum>QSizePolicy::Expanding</enum></property>
                 </spacer>
                </item>
                <item>
                 <widget class="QPushButton" name="connectExistingRepoButton">
                  <property name="text"><string>Connect</string></property>
                  <property name="toolTip"><string>Connect to an existing Git remote repository.</string></property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="configuredRepoWidget" native="true">
          <!-- This widget will be hidden by default in Python code -->
          <layout class="QVBoxLayout" name="configuredRepoLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QGroupBox" name="repoDetailsGroupBox">
             <property name="title">
              <string>Repository Details</string>
             </property>
             <layout class="QHBoxLayout" name="repoDetailsLayout">
              <item>
               <widget class="QLabel" name="repoPathLabel">
                <property name="text">
                 <string>Path:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="repoPathLineEdit">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
                <property name="placeholderText">
                 <string>N/A</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="changeRepoButton">
                <property name="text">
                 <string>Change/Reconfigure...</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="repoActionsGroupBox">
             <property name="title">
              <string>Repository Actions</string>
             </property>
             <layout class="QHBoxLayout" name="repoActionsLayout">
              <item>
               <spacer name="repoActionsSpacerLeft">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Expanding</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QPushButton" name="pullRepoButton">
                <property name="text">
                 <string>Pull</string>
                </property>
                <property name="toolTip">
                 <string>Fetch and integrate the latest changes from the remote repository.</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="pushRepoButton">
                <property name="text">
                 <string>Push</string>
                </property>
                <property name="toolTip">
                 <string>Upload local commits to the remote repository.</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="commitRepoButton">
                <property name="text">
                 <string>Commit...</string>
                </property>
                <property name="toolTip">
                 <string>Stage and commit local changes.</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_backupSharing">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>